### ISSUE TITLE
Add helper function to unify how we subsample data

### ISSUE THREAD

**Issue by isaac-chung** [embeddings-benchmark/mteb#508]
Piggy-backing off @imenelydiaker 's [comment](https://github.com/embeddings-benchmark/mteb/pull/424#issuecomment-2063881255):

When the dataset split is beyond 2048, subsampling is often used in this manner without stratification by label:
```python
self.dataset["test"] = (
    self.dataset["test"].shuffle(seed=self.seed).select(range(N_SAMPLES))
)
```
I've primarily seen this in classification datasets, but I imagine this would apply to other tasks as well.

A helper function can be introduced to help ease the contribution efforts when dealing with larger datasets, and to help unify how we subsample data. Example usage:
```python
self.dataset['test'] = stratified_subsampling(self.dataset['test'], N_SAMPLES)
```

**Comment by imenelydiaker** [comment]
Totally agree, we can use `train_test_split` from `datasets` ? Or we can create a the core of the function. There may exist better options here.

I'd add a variable to choose on what to stratify:
```py
self.stratified_subsampling(self.dataset, label, n_samples)
```
Should the method be defined in `AbsTask`?

**Comment by imenelydiaker** [comment]
We should totally point new contributors towards this yes!

Although I think it's posible to replace old subsampling functions, there are not a lot of them for the moment. We also should be able to run evals again on `multilingual-e5-small` and `paraphrase-multilingual-MiniLM-L12-v2`.

If you want we can split the work, you can open a PR with the new subsampling function proposition, and I'll update tasks using the old version and run evaluations again, wdyt?

**Comment by isaac-chung** [comment]
Ya let's do that 💪

**Comment by KennethEnevoldsen** [comment]
Wonderful suggestion @isaac-chung and @imenelydiaker!
